source/common/config/utility.h

```diff
@@ -531,7 +531,47 @@ class Utility {
     }
     return std::make_unique<FixedBackOffStrategy>(dns_refresh_rate_ms);
   }
-};
 
+  /**
+   * Prepares Jittered Exponential BackOff Strategy from config containing the Retry Policy
+   * @param config config containing RetryPolicy <envoy_v3_api_msg_config.core.v3.RetryPolicy>
+   * @param random random generator
+   * @param default_base_interval_ms (optional) Default base interval
+   * @param default_max_interval_ms (optional) Default maximum interval
+   * @return JitteredExponentialBackOffStrategyPtr if 1. RetryPolicy containing backoff values is
+   * found in config or 2. Default base interval and Default maximum interval is specified, else
+   * nullptr
+   */
+  template <typename T>
+  static JitteredExponentialBackOffStrategyPtr prepareJitteredExponentialBackOffStrategy(
+      const T& config, Random::RandomGenerator& random,
+      absl::optional<const uint32_t> default_base_interval_ms,
+      absl::optional<const uint32_t> default_max_interval_ms) {
+    // 1. RetryPolicy containing backoff values is found in config
+    if (config.has_retry_policy() && config.retry_policy().has_retry_back_off()) {
+      return prepareJitteredExponentialBackOffStrategy(config.retry_policy().retry_back_off(),
+                                                       random);
+    }
+
+    // 2. Default base interval and Default maximum interval is specified
+    if (default_base_interval_ms != absl::nullopt && default_max_interval_ms != absl::nullopt) {
+      return std::make_unique<JitteredExponentialBackOffStrategy>(
+          default_base_interval_ms.value(), default_max_interval_ms.value(), random);
+    }
```

Comment:
Looking at this again, I think this function can be further simplified. My goal is that this function will always return a valid strategy (non-nullptr), so its callers don't need to handle corner cases.
If the idea is that the caller will always pass a `default_base_interval_ms`, then the default max-interval will always be 10. WDYT?

Reply:
the problem with always returning a valid strategy is that the default values in this case are in `grpc_stream.h`. but the config is accessible only in `subscription_factory_impl.cc`. 
We would have to move the defaults to `subscription_factory_impl.cc` but I am not sure if moving them from `grpc_stream.h`. is a good idea.

Reply:
> the problem with always returning a valid strategy is that the default values in this case are in `grpc_stream.h`. but the config is accessible only in `subscription_factory_impl.cc`. We would have to move the defaults to `subscription_factory_impl.cc` but I am not sure if moving them from `grpc_stream.h`. is a good idea.

Why should they stay in `grpc_stream.h`? I may be missing something, so feel free to explain a bit here.
FWIW, I think the [original PR](https://github.com/envoyproxy/envoy/pull/1124) had the default value in something that is more similar to the subscription factory that we have at the moment.

Reply:
thanks for pointing out, I have moved the defaults to `envoy/config/subscription_factory.h`